disruption_py/machine/cmod/physics.py

```
@@ -591,24 +604,44 @@ def _get_power(times, p_lh, t_lh, p_icrf, t_icrf, p_rad, t_rad, p_ohm):
             # shots, excluding times for
             # which p_rad (uncalibrated) <= 1.e5 W
             dprad = np.gradient(p_rad, t_rad)
-            p_rad = interp1(t_rad, p_rad, times)
+            p_rad = interp1(t_rad, p_rad, times, fill_value=0)
             dprad = interp1(t_rad, dprad, times)
+
+        # Replace nans with zeros in p_ohm
+        p_ohm = np.nan_to_num(p_ohm, nan=0.0)
+
+        # Calculate radiated fraction
         p_input = p_ohm + p_lh + p_icrf
         rad_fraction = p_rad / p_input
         rad_fraction[rad_fraction == np.inf] = np.nan
+
+        # Calculate radiation confinement time
+        wmhd = interp1(efit_time, wmhd, times)
+        with np.errstate(divide="ignore", invalid="ignore"):
+            tau_rad = wmhd / p_rad
+        tau_rad[tau_rad == np.inf] = np.nan
```

Minor: instead of 
```python
with np.errstate(divide="ignore", invalid="ignore"):
    tau_rad = wmhd / p_rad
tau_rad[tau_rad == np.inf] = np.nan
```
we can do
```python
tau_rad = np.full_like(wmhd, np.nan)
np.divide(wmhd, p_rad, out=tau_rad, where=p_rad != 0)
```
to more explicitly handle divisions by zero. This ensures we only perform the division where `p_rad` is non-zero and anywhere we would have divided by is a nan instead of inf. I like this better than ignoring the errors because it is more explicit, but lmk if you prefer the `with` statement

---

no offense, but that's ugly! :chipmunk: 
what about something more physics-based, but still explicit, like so?
```python
tau_rad = wmhd / np.where(p_rad != 0, p_rad, np.nan)
```
that should be sufficient by itself. LMK!